src/protocol/codecs/algomsg.rs

```diff
@@ -0,0 +1,155 @@
+use std::io;
+
+use bytes::BytesMut;
+use tokio_util::codec::Decoder;
+use tracing::*;
+
+use super::topic::TopicCodec;
+use crate::protocol::invalid_data;
+
+/// [Tag] represents a message type identifier.
+///
+/// The original tag list can be found in go-algorand/protocol/tags.go.
+#[derive(Debug)]
+pub enum Tag {
+    UnknownMsg,
+    AgreementVote,
+    MsgOfInterest,
+    MsgDigestSkip,
+    NetPrioResponse,
+    Ping,
+    PingReply,
+    ProposalPayload,
+    StateProofSig,
+    TopicMsgResp,
+    Txn,
+    UniCatchupReq,
+    UniEnsBlockReq,
+    VoteBundle,
+}
+
+impl TryFrom<BytesMut> for Tag {
+    type Error = io::Error;
+
+    fn try_from(tag: BytesMut) -> Result<Self, Self::Error> {
+        let tag = std::str::from_utf8(&tag)
+            .map_err(|_| invalid_data!("couldn't convert the tag to a utf8 string"))?;
+
+        Ok(match tag {
+            "??" => Self::UnknownMsg,
+            "AV" => Self::AgreementVote,
+            "MI" => Self::MsgOfInterest,
+            "MS" => Self::MsgDigestSkip,
+            "NP" => Self::NetPrioResponse,
+            "pi" => Self::Ping,
+            "pj" => Self::PingReply,
```

Comment:
is this just inconsistent naming on Algorand side? just making sure that capitalization doesn't just differ "at random", in which case we'd be better off making our comparison case-insensitive

Reply:
> is this just inconsistent naming on Algorand side?

yep 🤷‍♂️